### PR TITLE
Updates to TbProfiler and Mykrobe parser

### LINKED ISSUE #15: tb-profiler: assigns wrong variant type
TBprofiler parser seems to only assign substitution variant type.

For instance the test file contains a `conservative_inframe_deletion` in `rpoB` is reported as a substitution.

### PR DESCRIPTION
**Pull request by mhkc**
This PR fixes 
- issues with the TbProfiler parser
- updates how phenotype information is stored for Mykrobe and TbProfiler
- clean up the Mykrober parser code a bit

close #15 #14

### CODE REVIEW

File: prp/models/phenotype.py [hunk 1]
```diff
@@ -150,11 +152,8 @@ class VariantBase(DatabaseReference):
     ref_aa: Optional[str] = None
     alt_aa: Optional[str] = None
     # prediction info
-    conf: Optional[int] = None
```
Comment (ryanjameskennedy) [inline prp/models/phenotype.py:153]: I reckon we still need the confidence score for mykrobe (the score out of ~20000)

File: prp/models/phenotype.py [hunk 2]
```diff
@@ -174,19 +173,25 @@ class VariantBase(DatabaseReference):
     res_subclass: Optional[str] = None
     method: Optional[str] = None
     close_seq_name: Optional[str] = None
-    type: Optional[str] = None
     change: Optional[str] = None
-    nucleotide_change: Optional[str] = None
-    protein_change: Optional[str] = None
-    annotation: Optional[List[Dict]] = None
-    drugs: Optional[List[Union[Dict, str]]] = None
```
Comment (ryanjameskennedy) [inline prp/models/phenotype.py:182]: Why no more drugs?

File: prp/models/phenotype.py [hunk 3]
```diff
@@ -174,19 +173,25 @@ class VariantBase(DatabaseReference):
     res_subclass: Optional[str] = None
     method: Optional[str] = None
     close_seq_name: Optional[str] = None
-    type: Optional[str] = None
     change: Optional[str] = None
-    nucleotide_change: Optional[str] = None
```
Comment (ryanjameskennedy) [inline prp/models/phenotype.py:179]: This and `protein_change` were things requested by Jonas

File: prp/parse/phenotype/tbprofiler.py [hunk 4]
```diff
@@ -43,36 +48,70 @@ def _get_tbprofiler_amr_sr_profie(tbprofiler_result):
...
             depth=hit["depth"],
-            freq=float(hit["freq"]),
+            frequency=float(hit["freq"]),
             ref_database=tbprofiler_result["db_version"]["name"],
             type=hit["type"],
             nucleotide_change=hit["nucleotide_change"],
             protein_change=hit["protein_change"],
             annotation=hit["annotation"],
-            drugs=hit["drugs"],
+            phenotype=hit["drugs"],
```
Comment (ryanjameskennedy) [inline prp/parse/phenotype/tbprofiler.py:79]: Why isn't this `phenotype` subjected to the same pydantic checks as mykrobe phenotype in `prp/parse/phenotype/mykrobe.py` line 75? One is in a list and the other not?

File: prp/models/phenotype.py [hunk 5]
```diff
@@ -174,19 +173,25 @@ class VariantBase(DatabaseReference):
     res_subclass: Optional[str] = None
     method: Optional[str] = None
     close_seq_name: Optional[str] = None
-    type: Optional[str] = None
```
Comment (ryanjameskennedy) [inline prp/models/phenotype.py:177]: Why no type?

**Review by ryanjameskennedy — CHANGES_REQUESTED** [review]

File: prp/parse/phenotype/mykrobe.py [hunk 6]
```diff
@@ -72,29 +71,47 @@ def _parse_mykrobe_amr_variants(mykrobe_result) -> Tuple[ResistanceVariant, ...]
...
+        # Mykrobe CSV variant format
+        # <gene>_<amino acid change>-<dna change>:<ref depth>:<alt depth>:<genotype confidence>
+        # ref: https://github.com/Mykrobe-tools/mykrobe/wiki/AMR-prediction-output
+        PATTERN = re.compile(
+            r"(.+)_(.+)-(.+):(\d+):(\d+):(\d+)", re.I
+        )
+        for variant in variants:
+            # extract variant info using regex
+            match = re.search(PATTERN, variant)
+            name, aa_change, dna_change, ref_depth, alt_depth, conf = match.groups()
```
[the remark this answers is not in the document]
Comment (mhkc) [reply prp/parse/phenotype/mykrobe.py:89]: changed to gene, from mykrobe doc

File: prp/models/phenotype.py (hunk 2 above)
Comment (mhkc) [reply prp/models/phenotype.py:182]: Drugs are bad for you

File: prp/models/phenotype.py (hunk 1 above)
Comment (mhkc) [reply prp/models/phenotype.py:153]: Renamed to `confidence`

File: prp/models/phenotype.py (hunk 5 above)
Comment (mhkc) [reply prp/models/phenotype.py:177]: whats the difference between `type`, `variant_type` and `method`

Comment (ryanjameskennedy) [reply prp/models/phenotype.py:177]: `variant_type` is substitution, deletion or insertion. `type` is the type of mutation missense, nonsense or frameshift. `method` is `genotype_model`.

Comment (mhkc) [reply prp/models/phenotype.py:177]: I renamed `type` to `variant_effect` to be more specific

File: prp/models/phenotype.py (hunk 3 above)
Comment (mhkc) [reply prp/models/phenotype.py:179]: Protein and DNA change is kept for TB profiler

File: prp/models/phenotype.py (hunk 5 above)
Comment (ryanjameskennedy) [reply prp/models/phenotype.py:177]: Great!

File: prp/models/phenotype.py (hunk 3 above)
Comment (ryanjameskennedy) [reply prp/models/phenotype.py:179]: And removed from mykrobe parser?

**Review by ryanjameskennedy — APPROVED** [review]
LGTM!

**Pull request merged**